upgrade/upgrade-instructions-for-improved-data-fixtures-and-tests.md

```diff
@@ -0,0 +1,63 @@
+Along with that, we also modified tests to make them resistant against such changes.
+
+Most of the changes were done in `project-base` so if you want them in your project after the upgrade, you need to do that manually.
+If any of the following instructions is not clear enough for you or you need more information, we recommend to take a look at the actual changes in [#1425](https://github.com/shopsys/shopsys/pull/1425).
+
+- Change the way the data fixtures are created:
+    - Remove all `Multidomain*DataFixture` classes and move the logic to regular `DataFixtures` classes.
+    - When setting multilanguage or multidomain attributes, use translator (`t` or `tc` function with "dataFixtures" domain) and [`Domain`](https://github.com/shopsys/shopsys/blob/v8.1.0/packages/framework/src/Component/Domain/Domain.php) service to fill the values for all existing domains and locales, e.g.:
+        ```php
+            foreach ($this->domain as $domainConfig) {
+                $locale = $domainConfig->getLocale();
+                $domainId = $domainConfig->getId();
+                $productData->name[$locale] = t('Name', [], 'dataFixtures', $locale);
+                $productData->description[$domainId] = t('Name', [], 'dataFixtures', $locale);
+            }
+        ```
+    - Remove `demo-data-products.csv` and `demo-data-customers.csv`, along with all the classes responsible for loading data from CSV files and move the data directly to [`ProductDataFixture`](https://github.com/shopsys/shopsys/blob/v8.1.0/project-base/src/Shopsys/ShopBundle/DataFixtures/Demo/ProductDataFixture.php) (you need to modify [`Performance/ProductDataFixture`](https://github.com/shopsys/shopsys/blob/v8.1.0/project-base/src/Shopsys/ShopBundle/DataFixtures/Performance/ProductDataFixture.php) as well) and [`UserDataFixture`](https://github.com/shopsys/shopsys/blob/v8.1.0/project-base/src/Shopsys/ShopBundle/DataFixtures/Demo/UserDataFixture.php).
+    - update your [`phpstan.neon`](https://github.com/shopsys/shopsys/blob/v8.1.0/project-base/phpstan.neon):
+        ```diff
+        -    # A helper methods returning an array of persistent references using $this->getReference()
+        -    message: '#^Method Shopsys\\ShopBundle\\DataFixtures\\ProductDataFixtureReferenceInjector::.+\(\) should return array<.+> but returns array<string, object>\.$#'
+        -    path: %currentWorkingDirectory%/src/Shopsys/ShopBundle/DataFixtures/ProductDataFixtureReferenceInjector.php
+        +    # Actually, we are setting an array item using "$array[] = $this->getReference()"
+        +    message: '#^Array \(array<.+>\) does not accept object\.$#'
+        +    path: %currentWorkingDirectory%/src/Shopsys/ShopBundle/DataFixtures/Demo/ProductDataFixture.php
+        ```
+    - Use [`PriceConverter`](https://github.com/shopsys/shopsys/blob/v8.1.0/packages/framework/src/Model/Pricing/PriceConverter.php) class for converting prices into all currencies, e.g.:
```

Comment:
isn't it used for converting to all domain default currencies?

Reply:
fixed in 3e53ddd53b6421b5025579dee4a29f0dcefefa85